### PR TITLE
Tebert/ticket detail

### PR DESCRIPTION
**Pull request by tebertt**
### Created TicketDetail, along with TicketEditDetail

**Ticket List**
- Fixed the navLink from ticketList to ticketDetail
- Fixed ticketList property wrapper

**TicketDetail/Edit**
- Title: is required, will display alert if empty and not update
- Assigned user: I wanted this initially to have a pop out menu with a few recommended options with a more button that would push to a new page with all the options. I was having trouble creating a custom context button in a picker. I got pretty far by using a menu instead but it wasn't conforming to the UX I wanted, so I decided to just to push to the new page without the menu. I'll look into implementing this at a different time.
- Statuses: ditto for assigned user
- Tags: Hidden navlink that pushes to a multi selection list. Selected items are appended, so the order displayed is the order selected
- Date: Toggle with animation, shows date picker when true
- Description: Larger textfield, had to use empty non-optional string in edit, easier to work with with textfield

**Other**
- Added custom nilContent option to nilContext (Known bug: does not like custom parameters on view components)
- Added multiple scene window to info.plist, this stopped debugger warning
- Added multi selection list for tags
- Added Extended picker for assigned users/statuses
- Added optional operator

### CODE REVIEW

File: Sluggo-SwiftUI/Views/Generic Views/ExtendedPicker.swift
Comment (agavgavi) [inline Sluggo-SwiftUI/Views/Generic Views/ExtendedPicker.swift]: If this is unnecessary it should be deleted.

File: Sluggo-SwiftUI/Views/TicketEditDetail.swift
Comment (agavgavi) [inline Sluggo-SwiftUI/Views/TicketEditDetail.swift:14]: If alertContext is used, it needs to be attached to the view with a call to:
`.alert(context: alertContext)`
otherwise present errors won't do anything

File: Sluggo-SwiftUI/Views/TicketListView.swift
Comment (agavgavi) [inline Sluggo-SwiftUI/Views/TicketListView.swift]: If unnecessary, delete

File: Sluggo-SwiftUI/Views/TicketEditDetail.swift
Comment (agavgavi) [inline Sluggo-SwiftUI/Views/TicketEditDetail.swift:117]: If we want to use this view to also deal with Creating tickets, we need to consider if this function is going to handle both cases or just update. More of a discussion rather than a change.

File: Sluggo-SwiftUI/Views/TicketEditDetail.swift
Comment (agavgavi) [inline Sluggo-SwiftUI/Views/TicketEditDetail.swift:25]: I feel like a good portion of the functions and variables of this view should be abstracted away to a viewModel. But I can do that after we merge this if you'd like.

File: Sluggo-SwiftUI/Views/Components/OptionalOperator.swift
Comment (agavgavi) [inline Sluggo-SwiftUI/Views/Components/OptionalOperator.swift:11]: What is this used for?

**Review by agavgavi — CHANGES_REQUESTED** [review]
Just some thoughts and questions.

File: Sluggo-SwiftUI/Views/Generic Views/ExtendedPicker.swift
Comment (tebertt) [reply Sluggo-SwiftUI/Views/Generic Views/ExtendedPicker.swift]: I planned on working on it a bit more. I'll move it to a separate branch

File: Sluggo-SwiftUI/Views/TicketEditDetail.swift
Comment (tebertt) [reply Sluggo-SwiftUI/Views/TicketEditDetail.swift:14]: I'll do that

File: Sluggo-SwiftUI/Views/TicketEditDetail.swift
Comment (tebertt) [reply Sluggo-SwiftUI/Views/TicketEditDetail.swift:117]: I'll see if I can abstract the view a bit more

File: Sluggo-SwiftUI/Views/Components/OptionalOperator.swift
Comment (tebertt) [reply Sluggo-SwiftUI/Views/Components/OptionalOperator.swift:11]: This is the ?? operand, used for passing optional generics, I use this specifically for optional ticket fields in text fields

**Review by agavgavi — APPROVED** [review]
LGTM

**Pull request merged**
